### PR TITLE
Fixed bash completions

### PR DESCRIPTION
**Pull request by dimitern**
Makefile's install-etc was not installing juju-version
Changed the completions to work for a juju binary, built
from source, as well as the one from the juju-2.0 package.

See bug http://pad.lv/1622775 for more details
(PR was intended to fix it, but simply improves CLI UX)

QA steps:
 1. make install-etc && source /etc/bash_completion
 2. juju boo[tab]tstrap
 3. juju de[tab]p[tab]loy ubuntu
 4. juju status [tab]u[tab]buntu/0

### CODE REVIEW

File: etc/bash_completion.d/juju-2.0 [hunk 1]
```diff
@@ -42,9 +42,11 @@ import json, sys
...
 j = json.load(sys.stdin)
 all_units = []
-for k, v in j.get("applications", {}).items():
-    all_units.extend(v.get("units", {}).keys())
-print ("\n".join([unit + trail for unit in all_units]))
+try:
+    for k, v in j.get("applications", {}).items():
+        all_units.extend(v.get("units", {}).keys())
+    print ("\n".join([unit + trail for unit in all_units]))
+except: pass
```
Comment (bz2) [inline etc/bash_completion.d/juju-2.0]: I think we don't want `try:/except: pass` - yes the kipple when things break is bad, but things being broken with no feedback as to why is worse I feel.

File: etc/bash_completion.d/juju-2.0 [hunk 2]
```diff
@@ -335,13 +339,26 @@ _juju_complete_2_0() {
 }
 # _juju_2_0_cache_TTL [mins]
 export _juju_2_0_cache_TTL=2
+
 # All above completion is juju-2.0 specific, uses $_juju_cmd_JUJU_2_0
-export _juju_cmd_JUJU_2_0=/usr/bin/juju-2.0
+
+# Detect juju built from source (highest priority)
+if [ -x "$GOPATH/bin/juju" ]; then 
+    export _juju_cmd_JUJU_2_0="$GOPATH/bin/juju"
```
Comment (bz2) [inline etc/bash_completion.d/juju-2.0]: This seems good. Might mean developers see packaging issues less, but in exchange will be catching their own problems instead.

File: Makefile [hunk 3]
```diff
@@ -99,7 +99,8 @@ endif
 # Install bash_completion
 install-etc:
 	@echo Installing bash completion
-	@sudo install -o root -g root -m 644 etc/bash_completion.d/juju2 /etc/bash_completion.d
+	@sudo install -o root -g root -m 644 etc/bash_completion.d/juju-2.0 /etc/bash_completion.d
+	@sudo install -o root -g root -m 644 etc/bash_completion.d/juju-version /etc/bash_completion.d
```
Comment (bz2) [inline Makefile]: These locations should probably be updated to the new `/usr/share/bash-completion/bash_completion` location. The split between /etc and there is also the cause of much confusion.

File: etc/bash_completion.d/juju-2.0 (hunk 2 above)
Comment (dimitern) [inline etc/bash_completion.d/juju-2.0]: Yeah, and makes completions work for devs as well as other users :)

File: Makefile (hunk 3 above)
Comment (dimitern) [inline Makefile]: :+1: Done.

**Comment by dimitern** [comment]
$$merge$$

**Pull request merged**